build.cake

```diff
@@ -1,145 +1,93 @@
     .Does(() =>
     {
-        NuGetRestore("ReactiveXComponent.sln", new NuGetRestoreSettings { NoCache = true });
-        BuildSolution(@"./ReactiveXComponent.sln", buildConfiguration, setAssemblyVersion, version);
+        DotNetCoreRestore("ReactiveXComponent.sln");
+        DotNetCoreBuild(
+        "ReactiveXComponent.sln",
+        new DotNetCoreBuildSettings {
+            Configuration = buildConfiguration,
+        });
     });
 
 Task("Test")
     .Does(() =>
     {
-        var testAssembliesPatterns = new string[]
+        var settings = new DotNetCoreTestSettings
         {
-            "./ReactiveXComponentTest/bin/" + buildConfiguration + "/ReactiveXComponentTest.dll"
+            Configuration = buildConfiguration
         };
-        
-        var testAssemblies = GetFiles(testAssembliesPatterns);
-        var nunitSettings = new NUnit3Settings(){ Results = new[] { new NUnit3Result { FileName = "TestResults.xml" } } };
-        NUnit3(testAssemblies, nunitSettings);
-    });
 
-Task("Merge")
-    .Does(() =>
-    {
-        EnsureDirectoryExists("packaging");
-
-        var filesToMerge = GetFiles("./ReactiveXComponent/bin/"+ buildConfiguration + "/*.dll");
-
-        var ilRepackSettings = new ILRepackSettings { Parallel = true, Internalize = true };
-
-        ILRepack(
-            "./packaging/ReactiveXComponent.dll",
-            "./ReactiveXComponent/bin/"+ buildConfiguration + "/ReactiveXComponent.dll",
-            filesToMerge,
-            ilRepackSettings
-		);
-
-        var pdbFiles = GetFiles("./ReactiveXComponent/bin/"+ buildConfiguration + "/ReactiveXComponent.pdb");
-        CopyFiles(pdbFiles, "./packaging");
+        var projectFiles = GetFiles("./**/*Test.csproj");
+        foreach(var file in projectFiles)
+        {
+            DotNetCoreTest(file.FullPath, settings);
+        }
     });
 
 Task("CreatePackage")
-    .IsDependentOn("Merge")
     .Does(() =>
     {
-        EnsureDirectoryExists("nuget");
-
-        var formattedNugetVersion = FormatNugetVersion(version);
-
-        var filesToPackPatterns = new string[]
-        {
-            "./packaging/*.dll",
-            "./packaging/*.pdb"
-        };
-
-        var filesToPack = GetFiles(filesToPackPatterns);
-
-        var nuSpecContents = filesToPack.Select(file => new NuSpecContent {Source = file.FullPath, Target = @"lib\net451"}).ToList();
-
-        var nugetPackSettings = new NuGetPackSettings()
-        { 
-            OutputDirectory = @"./nuget",
-            Files = nuSpecContents,
-            Version = formattedNugetVersion,
-            IncludeReferencedProjects = true
-        };
-
-        NuGetPack("ReactiveXComponent.Net.nuspec", nugetPackSettings);
+        DotNetCorePack(
+            "ReactiveXComponent/ReactiveXComponent.csproj",
+            new DotNetCorePackSettings  {
+                Configuration = buildConfiguration,
+                OutputDirectory = @"nuget",
+                VersionSuffix = packageVersion,
```

Comment:
We should use the same versioning logic (with formattedNugetVersion)

Reply:
update with : var formattedNugetVersion = FormatNugetVersion(version);